### PR TITLE
feature/EN-83

### PR DESCRIPTION
**Pull request by yard**

### CODE REVIEW

File: src/classes/BroadPhases/BasicPooledBroadphase.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                var body = objects[ j ];
+
+                // first test AABB intersection (~ broad phase)
+                if ( body.aabb.testRayIntersect( _start, _end ) ) {
+                    // if AABB intersects, as the body about inner intersections
+                    body.rayIntersect( _start, _end, limit, intersections );
+                }
+
+                if ( limit && ( intersections.length >= limit ) ) {
+                    return intersections.slice( 0, limit );
```
Comment (stepan-beresnev) [inline src/classes/BroadPhases/BasicPooledBroadphase.js:133]: No need to sort?

File: src/classes/BroadPhases/BasicPooledBroadphase.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                }
+            }
+        }
+
+        intersections.sort( function ( a, b ) {
```
Comment (stepan-beresnev) [inline src/classes/BroadPhases/BasicPooledBroadphase.js:138]: Comparison function can be declared in outer scope, just to keep memory

File: src/classes/BroadPhases/BasicPooledBroadphase.js (hunk 1 above)
Comment (yard) [reply src/classes/BroadPhases/BasicPooledBroadphase.js:133]: Technically yes, but in fact we need to re-implement raycasts to properly find first and closest hit

File: src/classes/BroadPhases/BasicPooledBroadphase.js (hunk 2 above)
Comment (yard) [reply src/classes/BroadPhases/BasicPooledBroadphase.js:138]: Let's postpone this up to raycast refactoring.

**Review by stepan-beresnev — APPROVED** [review]

**Pull request merged**